src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -612,13 +612,17 @@ void visitFunctionPointerSignature(IMethodSymbol symbol)
                 foreach (var param in symbol.Parameters)
                 {
                     // https://github.com/dotnet/roslyn/issues/61647: Use public API.
-                    Debug.Assert((param as Symbols.PublicModel.ParameterSymbol)?.GetSymbol<ParameterSymbol>().EffectiveScope switch
+#if DEBUG
+                    if ((param as Symbols.PublicModel.ParameterSymbol)?.GetSymbol<ParameterSymbol>() is var p)
```

Comment:
`var p` will always match, even if the input is null, so it seems strange to immediately deref it in `p.EffectiveScope`, as well as to use `as`-casts and `?.` in order to get the internal parameter symbol here.

Basically I would expect something like `var p = ((Symbols.PublicModel.ParameterSymbol)param).GetSymbol<ParameterSymbol>(); Debug.Assert(...);` here.